pkg/webhook/admission/response.go

```diff
@@ -107,3 +107,10 @@ func validationResponseFromStatus(allowed bool, status metav1.Status) Response {
 	}
 	return resp
 }
+
+// WithWarnings adds the given warnings to the Response.
+// If any warnings were already given, they will not be overwritten.
+func (r Response) WithWarnings(warnings []string) Response {
```

Comment:
variadic would have better ergonomics here, I believe

Reply:
Oh yeah, that's a great suggestion! Will update